src/wp-includes/option.php

```diff
@@ -253,6 +253,110 @@ function get_option( $option, $default_value = false ) {
 	return apply_filters( "option_{$option}", maybe_unserialize( $value ), $option );
 }
 
+/**
+ * Primes specific options into the cache with a single database query.
+ *
+ * Only options that do not already exist in cache will be primed.
+ *
+ * @since 6.4.0
+ *
+ * @global wpdb $wpdb WordPress database abstraction object.
+ *
+ * @param array $options An array of option names to be primed.
+ */
+function prime_options( $options ) {
+	$alloptions     = wp_load_alloptions();
+	$cached_options = wp_cache_get_multiple( $options, 'options' );
+
+	// Filter options that are not in the cache.
+	$options_to_prime = array();
+	foreach ( $options as $option ) {
+		if ( ( ! isset( $cached_options[ $option ] ) || ! $cached_options[ $option ] ) && ! isset( $alloptions[ $option ] ) ) {
+			$options_to_prime[] = $option;
+		}
+	}
+
+	// Bail early if there are no options to be primed.
+	if ( empty( $options_to_prime ) ) {
+		return;
```

Comment:
A coverage report shows that this line isn't hit by the current tests:
![image](https://github.com/WordPress/wordpress-develop/assets/79332690/ab79e3a9-7b5b-4888-8923-2b11f90b9a2f)

Let's add a test that hits this line and that no additional queries run.

Reply:
Thanks, Added in https://github.com/WordPress/wordpress-develop/pull/4970/commits/239fa6bc058dc1eb07e8f79f0418f89383b5af70